### PR TITLE
Bmi055 temperature fix

### PR DESCRIPTION
**Pull request by dmillard-TD**
Fixes issue #11854 
Discussion of this bug is here:
http://discuss.px4.io/t/pixhawk-4-temperature-compensation-failure

Problem- the BMI055 IMU driver was not reporting temperature correctly, making it impossible to perform temperature compensation on this sensor.

Fixed it by:
 -modified the register read value to be stored in an int8_t instead of uint8_t, in order to preserve the sign for values less than zero
 -Perform scaling and offset according to the BMI055 datasheet before populating the report buffer

Test performed in an industrial environmental chamber run from -40 C to +60 C.  Results now match the temperature reported by the other IMU (ICM20689) and the temperature compensation script runs without errors.

### CODE REVIEW

File: src/drivers/imu/bmi055/BMI055_accel.cpp [hunk 1]
```diff
@@ -523,7 +523,7 @@ BMI055_accel::measure()
 		int16_t     accel_x;
 		int16_t     accel_y;
 		int16_t     accel_z;
-		int16_t     temp;
```
Comment (LorenzMeier) [inline src/drivers/imu/bmi055/BMI055_accel.cpp]: This was wrong in the driver originally.

File: src/drivers/imu/bmi055/BMI055_gyro.cpp [hunk 2]
```diff
@@ -562,8 +562,7 @@ BMI055_gyro::measure()
 
 	check_registers();
 
-	uint8_t temp = read_reg(BMI055_ACC_TEMP);
-
+	int8_t temp = read_reg(BMI055_ACC_TEMP);
```
Comment (LorenzMeier) [inline src/drivers/imu/bmi055/BMI055_gyro.cpp]: This introduces load but is acceptable for now.

**Comment by LorenzMeier** [comment]
@dmillard-TD I've open-loop changed one thing: I've fixed the temperature read to be a single transfer. Ideally, the gyro temp would be copied from the accel driver, instead of reading it from the bus, as every transfer has a fixed overhead.

Could you please re-test on hardware and fix the CI errors that were still present?

On a sidenote: The code quality of this driver doesn't reach common PX4 standards and it needs love. I missed that on the first review when it was contributed, but the OA clearly had a limited understanding of integer arithmetic on an embedded system and did things manually that happen by themselves.

**Comment by dmillard-TD** [comment]
Fixed the formatting, it should pass CI now.  I'm starting a new test on the hardware, will update when it's complete.

**Comment by dmillard-TD** [comment]
Test is complete, it passed.  Should be ready to merge.

@LorenzMeier I am evaluating several sensors now, but when I am finished I'll try to circle back and give this driver the love it deserves.  :)

**Pull request merged**